### PR TITLE
Always add a space for multiple curried args invocation

### PR DESCRIPTION
**Pull request by janus**
Addresses https://github.com/dotnet/docs/pull/28141

### CODE REVIEW

**Comment by nojaf** [comment]
Hello, thank you for this contribution.

Could you please first open an issue in Fantomas instead of adding a link to a PR to the docs.
This helps for the release notes later on.

Next, please target the `4.7` branch instead of the master.
As you could not have known this, I've added some additional notes in https://github.com/fsprojects/fantomas/pull/2078.

File: src/Fantomas/CodePrinter.fs [hunk 1]
```diff
@@ -3215,6 +3215,10 @@ and genApp astContext e es ctx =
                 (fun ctx ->
                     match es with
                     | [] -> false
+                    | Paren _ :: rest when es.Length > 1 ->
```
Comment (nojaf) [inline src/Fantomas/CodePrinter.fs:3218]: This is wrong, as soon as there are more than elements in `es` it should return true.
Unit test ```maintain indent if when condition is multiline``` should be updated as well.

**Review by nojaf — CHANGES_REQUESTED** [review]
```fsharp 
       | AppWithLambda (e, es, lpr, lambda, rpr, pr) ->
```
should be updated as well to reflect these changes.
Something like:

```fsharp
            let sepSpaceAfterFunctionName =
                let sepSpaceBasedOnSetting e =
                    match e with
                    | Paren _ -> sepSpace
                    | UppercaseSynExpr -> (fun ctx -> onlyIf ctx.Config.SpaceBeforeUppercaseInvocation sepSpace ctx)
                    | LowercaseSynExpr -> (fun ctx -> onlyIf ctx.Config.SpaceBeforeLowercaseInvocation sepSpace ctx)

                match es with
                | [] -> sepSpaceBasedOnSetting e
                | _ -> sepSpace
```

File: src/Fantomas/CodePrinter.fs (hunk 1 above)
Comment (janus) [reply src/Fantomas/CodePrinter.fs:3218]: @nojaf 
It returns true if there are at least two curried args. It this fails, it continues with old logic.

Comment (nojaf) [reply src/Fantomas/CodePrinter.fs:3218]: There is no need for the old logic, that is my point.

**Comment by knocte** [comment]
Addressed comments in PR2077 (https://github.com/fsprojects/fantomas/pull/2077) that supersedes this one.

**Comment by knocte** [comment]
err, PR2078

**Pull request closed without merging**